sql/src/main/java/io/crate/expression/operator/LikeOperator.java

```diff
@@ -60,33 +67,48 @@ public FunctionInfo info() {
             if (value == null) {
                 return this;
             }
-            return new CompiledLike(info, (String) value);
+            boolean ignoreCase = ((Literal<Boolean>) arguments.get(2)).value();
+            return new CompiledLike(info, (String) value, ignoreCase);
         }
         return super.compile(arguments);
     }
 
     @Override
     public Boolean evaluate(TransactionContext txnCtx, Input<String>... args) {
         assert args != null : "args must not be null";
-        assert args.length == 2 : "number of args must be 2";
+        assert args.length == 3 : "number of args must be 3";
 
         String expression = args[0].value();
         String pattern = args[1].value();
         if (expression == null || pattern == null) {
             return null;
         }
+        boolean ignoreCase = (Boolean) ((Object) args[2].value());
```

Comment:
Isn't the object cast redundant here?

Reply:
The value comes from ```Input<String>```, so it would be a String, but in reality it is a Boolean. I considered the possibility of using a different mechanism to pass the boolean, like metadata, i.e. functions can receive metadata in addition to arguments. But I thought that would look strange or a hack. I thought about making the inner type an Object, but then I have to cast more. In the end I settled for this.